internal/decryption/decryption.go

```diff
@@ -39,8 +41,10 @@ type Decryption struct {
 
 type Option func(d *Decryption)
 
-func NewDecryption(options ...Option) *Decryption {
-	d := &Decryption{}
+func NewDecryption(fs afero.Fs, options ...Option) *Decryption {
+	d := &Decryption{
+		opts: KeyProviderOpts{Fs: fs},
+	}
```

Comment:
the file system should be an option not a new parametter, this defets the purpose of having the options approach

Reply:
good point, thanks for the suggestion!
I made the filesystem an option that is applied using `func WithFilesystem`